addons/mrp/models/mrp_production.py

```diff
@@ -1740,7 +1740,7 @@ def _action_cancel(self):
         finish_moves = self.move_finished_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
         raw_moves = self.move_raw_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
         (finish_moves | raw_moves)._action_cancel()
```

Comment:
avoid to re-cancel self
```suggestion
        (finish_moves | raw_moves).with_context(skip_mo_check=True)._action_cancel()
```